### PR TITLE
[Doc] Move README to CLI

### PR DESCRIPTION
**Pull request by Eleven1Liu**
**Description**
1. Move README to chapter`Command Line Interface`. 
2. Add options (hidden).

### CODE REVIEW

File: docs/cli/cli.rst [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+* Clone `LibMultiLabel <https://github.com/ASUS-AICS/LibMultiLabel>`_.
+* Install the latest development version, run:
+
+.. code-block:: bash
+
+    pip3 install -r requirements.txt
+
+.. _cli-quickstart:
+
+Using CLI via an example
```
Comment (cjlin1) [inline docs/cli/cli.rst:30]:
```suggestion
Using CLI via an Example
```

File: docs/cli/cli.rst [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+See `Dataset Formats <data_format.html>`_ here if you want to use your own dataset.
+
+Step 2. Training and Prediction
+-------------------------------
+
+Train a CNN model and predict the test set by an example config. Use ``--cpu`` to run the program on the cpu.
+
+.. code-block:: bash
+
+    python3 main.py --config example_config/rcv1/kim_cnn.yml
```
Comment (cjlin1) [inline docs/cli/cli.rst:66]: need an example for linear here; can use also use rcv1

File: docs/cli/data_format.rst [hunk 3]
```diff
@@ -50,7 +10,7 @@ both :ref:`libmultilabel-format` and
 .. _libmultilabel-format:
 
 LibMultiLabel Format
-^^^^^^^^^^^^^^^^^^^^
+--------------------
 
 The LibMultiLabel format is a format for labels,
```
Comment (cjlin1) [inline docs/cli/data_format.rst:15]: for IDs (optional), labels, and raw texts.

File: docs/cli/data_format.rst [hunk 4]
```diff
@@ -50,7 +10,7 @@ both :ref:`libmultilabel-format` and
...
 raw texts and optional IDs. It combines them
```
Comment (cjlin1) [inline docs/cli/data_format.rst:16]: They are combined in a single file with tables ... It must satisfies the following requirements
- one ...
- ID, ... are separated by ...
- labels ...
- each field should not ... <TAB>

An example with the ...

...

An example without the ...

File: docs/cli/data_format.rst [hunk 5]
```diff
@@ -75,7 +35,7 @@ Without ID column::
 .. _libsvm-format:
 
 LibSVM Format
-^^^^^^^^^^^^^
+-------------
 
 The LibSVM format is a format for labels and sparse numerical
 features. It combines them in a single file,
```
Comment (cjlin1) [inline docs/cli/data_format.rst:41]: make similar changes as the libmultilabel part

Comment (cjlin1) [reply docs/cli/data_format.rst:41]: why not saying that

The format of each line is

<label1>,<label2>...,<labelL> <index1>:<value1> <index2>:<value2> ...

File: docs/cli/nn.rst [hunk 6]
```diff
@@ -0,0 +1,93 @@
+Training, Prediction, and Hyperparameter Search for Neural Networks
+===================================================================
+
+Training
```
Comment (cjlin1) [inline docs/cli/nn.rst:4]:
```suggestion
Training and (Optional) Prediction
```
as this part can do prediction as well

File: docs/cli/linear.rst [hunk 7]
```diff
@@ -0,0 +1,27 @@
+Training and Prediction for Linear Classifiers
+==============================================
+
+Training
```
Comment (cjlin1) [inline docs/cli/linear.rst:4]:
```suggestion
Training and (Optional) Prediction
```

File: docs/cli/linear.rst [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+
+For training, use
+
+.. code-block:: bash
+
+    python3 main.py --config CONFIG_PATH \
+                    --linear [--liblinear_options LIBLINEAR_OPTIONS]
+
+- **config**: configure parameters in a yaml file.
+
```
Comment (cjlin1) [inline docs/cli/linear.rst:15]: what if validation set specified in config? Mention that validation set isn't used

File: docs/cli/nn.rst [hunk 9]
```diff
@@ -0,0 +1,93 @@
...
+
+- **vocab_file**: set the file path to a predefined vocabulary set that contains lines of words.
+
+.. code-block::
+
+    the
+    a
+    an
+
+For the validation process in the training procedure, you can evaluate the model with a set of evaluation metrics. Set ``monitor_metrics`` to define what you want to print on the screen. The argument ``val_metric`` is the metric for picking the best model. Example:
```
Comment (cjlin1) [inline docs/cli/nn.rst:40]: The model occurred at the epoch with the best validation metric is returned after training. If  users did not specify a validation set in the configuration file or via --??, we ?? (mention the 80-20 split?)

File: docs/cli/data_format.rst [hunk 10]
```diff
@@ -1,47 +1,7 @@
...
-
-Data Format
------------
-
-To use the command line tool, the input data must have specific
-formats.
+The input data for building train, test, and validation datasets must have specific formats.
 For neural networks, the only accepted format is the
 :ref:`libmultilabel-format`. For linear methods,
 both :ref:`libmultilabel-format` and
```
Comment (cjlin1) [inline docs/cli/data_format.rst:7]: both ... are accepted.

File: docs/cli/data_format.rst [hunk 11]
```diff
@@ -5,29 +5,29 @@ The input data for building train, test, and validation datasets must have speci
...
-:ref:`libsvm-format` is accepted.
+:ref:`libsvm-format` are accepted.
 
 .. _libmultilabel-format:
```
Comment (Eleven1Liu) [inline docs/cli/data_format.rst:10]:
```suggestion

aaa test
```

File: docs/cli/data_format.rst [hunk 12]
```diff
@@ -5,29 +5,29 @@ The input data for building train, test, and validation datasets must have speci
...
 
 LibMultiLabel Format
 --------------------
 
-The LibMultiLabel format is a format for labels,
-raw texts and optional IDs. It combines them
-in a single file, using tabs and line endings as control characters.
-It has
+The LibMultiLabel format is a format for IDs (optional),
```
Comment (Eleven1Liu) [inline docs/cli/data_format.rst:15]:
```suggestion
askdfla;sdf
```

File: docs/cli/nn.rst [hunk 13]
```diff
@@ -0,0 +1,98 @@
...
+    the
+    a
+    an
+
+For validation, you can evaluate the model with a set of evaluation metrics.
+Set ``monitor_metrics`` to define what you want to print on the screen.
+The argument ``val_metric`` is the metric for selecting the best model.
+Namely, the model occurred at the epoch with the best validation metric is returned after training.
+If you do not specify a validation set in the configuration file via ``val_path`` or training-validation split ratio via ``val_size``,
+we will split the training data into training and validation set with an 80-20 split.
```
Comment (cjlin1) [inline docs/cli/nn.rst:45]:
```suggestion
we will split the training data into training and validation set with an 80-20 split. Example lines in a configuration file:
```

File: docs/cli/nn.rst [hunk 14]
```diff
@@ -0,0 +1,98 @@
...
+
+.. code-block:: yaml
+
+    monitor_metrics: [P@1, P@3, P@5]
+    val_metric: P@1
+
+
+If ``test_path`` is specified or ``DATA_DIR/test.txt`` exists, the model with the highest ``val_metric`` will be evaluated after training.
```
Comment (cjlin1) [inline docs/cli/nn.rst:53]:
```suggestion
If ``test_path`` is specified or ``DATA_DIR/test.txt`` exists, the model with the highest ``val_metric`` will be used to predict the test set.
```

**Review by cjlin1 — APPROVED** [review]

**Pull request merged**